plone/app/vocabularies/catalog.py

```diff
@@ -646,6 +659,181 @@ def __call__(self, context, query=None):
+
+      >>> context = create_context()
+
+      >>> catalog = DummyCatalog(('/1234', '/2345'))
+      >>> context.portal_catalog = catalog
+
+      >>> tool = DummyTool('portal_url')
+      >>> def getPortalPath():
+      ...     return '/'
+      >>> tool.getPortalPath = getPortalPath
+      >>> context.portal_url = tool
+
+      >>> vocab = StaticCatalogVocabulary({'portal_type': ['Document']})
+      >>> vocab
+      <plone.app.vocabularies.catalog.StaticCatalogVocabulary object at ...>
+
+      >>> vocab.search('')
+      <zope.schema.vocabulary.SimpleVocabulary object at ...>
+      >>> list(vocab.search(''))
+      []
+
+      >>> vocab.search('foo')
+      <zope.schema.vocabulary.SimpleVocabulary object at ...>
+
+      >>> [(t.title, t.value) for t in vocab.search('foo')]
+      [('BrainTitle (/1234)', '/1234'), ('BrainTitle (/2345)', '/2345')]
+
+    We strip out the site path from the rendered path in the title template:
+
+      >>> catalog = DummyCatalog(('/site/1234', '/site/2345'))
+      >>> context.portal_catalog = catalog
+      >>> vocab = StaticCatalogVocabulary({'portal_type': ['Document']})
+      >>> [(t.title, t.value) for t in vocab.search('bar')]
+      [('BrainTitle (/site/1234)', '/site/1234'),
+       ('BrainTitle (/site/2345)', '/site/2345')]
+
+      >>> context.__name__ = 'site'
+      >>> vocab = StaticCatalogVocabulary({'portal_type': ['Document']})
+      >>> [(t.title, t.value) for t in vocab.search('bar')]
+      [('BrainTitle (/1234)', '/site/1234'),
+       ('BrainTitle (/2345)', '/site/2345')]
+
+    The title template can be customized:
+
+      >>> vocab.title_template = "{url} {brain.UID} - {brain.Title} {path}"
+      >>> [(t.title, t.value) for t in vocab.search('bar')]
+      [('proto:/site/1234 /site/1234 - BrainTitle /1234', '/site/1234'),
+       ('proto:/site/2345 /site/2345 - BrainTitle /2345', '/site/2345')]
+
+    """
+    title_template = "{brain.Title} ({path})"
+    text_search_index = "SearchableText"
+
+    def __init__(self, query, text_search_index=None,
+                 title_template=None):
+        self.query = query
+        if text_search_index:
+            self.text_search_index = text_search_index
+        if title_template:
+            self.title_template = title_template
+
+    @property
+    @memoize
+    def nav_root_path(self):
+        site = getSite()
+        nav_root = getNavigationRootObject(site, site)
+        return '/'.join(nav_root.getPhysicalPath())
+
+    def get_brain_path(self, brain):
+        nav_root_path = self.nav_root_path
+        path = brain.getPath()
+        if path.startswith(nav_root_path):
+            path = path[len(nav_root_path):]
+        return path
+
+    @property
+    @view_memoize
+    def brains(self):
+        return self.catalog(**self.query)
+
+    def createTerm(self, brain, context=None):
+        return SimpleTerm(
+            value=brain.UID, token=brain.UID,
+            title=self.title_template.format(
```

Comment:
If I am not wrong this might break in Python 2, where title is expected to be text. Please consider this before merging

Reply:
You're right, we probably need some special handling here for Python 2 where `brain.Title` is always an encoded string, but `Term.title` is always unicode.

Reply:
This is the last blocker here before we can merge, right?

Reply:
Yeah, wrapping this with `safe_unicode` seem to be the right thing to do